QENSmodels/lorentzian.py

```diff
@@ -75,14 +75,35 @@ def lorentzian(x, scale=1.0, center=0.0, hwhm=1.0):
       | ``hwhm``         | FWHM /2         | W/2              |
       +------------------+-----------------+------------------+
 
+    * Numerical issues:
+      The definition of the Lorentzian function used here is such that
+      the integral between :math:`-\infty` and :math:`-\infty` is 1.
+      However, when the hwhm is comparable or smaller than the x step,
+      the sampling of the function will result in a numerical integral > 1.
+      E.g., in the extreme case where hwhm tends to zero, the numerical
+      sampling at x points will result in a delta-like function, but with
+      a value at maximum approaching :math:`-\infty` instead of the
+      value 1/:math:`\Delta x` used in the definition of the delta function.
+      Therefore, the value of the integral of the function is checked
+      and used to renormalize the returned function whenever the integral
+      is larger than 1.
+
     """
     # Input validation
     x = np.asarray(x)
 
     if hwhm == 0:
         model = QENSmodels.delta(x, scale, center)
     else:
-        model = scale * hwhm / ((x - center) ** 2 + hwhm ** 2) / np.pi
+        model = hwhm / ((x - center) ** 2 + hwhm ** 2) / np.pi
+
+    # Area normalization
+    area = np.trapz(model, x)
```

Comment:
This raises an error if `x` is a single number.

Reply:
Right. This is quite a mysterious to me. I did not think about passing a single number, and then I thought that the conversion of the float to a numpy array done in the input validation should handle this, but I just found that doing:
x = 0.0
x = np.asarray(x)
print(x.shape, x.size)
--> () 1
On the other hand:
x = [0.0]
x = np.asarray(x)
print(x.shape, x.size)
--> (1,) 1
The latter is accepted by numpy.trapz, returning an area of 0.

As it does not make sense to try to compute the area of a single point, my suggestion is to add a test before trying the area normalization, e.g.:
if x.size > 1:
    area = np.trapz(model, x)
    if area > 1: ...

Of course, your remark raises the question of what to do when the notion of area makes no real sense, e.g. when you pass a single point or a range of x values that do not provide a good representation of the function, e.g. if for a Lorentzian I pass the range [2,5] that does not include the peak. However, in all those cases I think that the value returned by trapz will be < 1, so in the end this should have no effect.

What do you think?

Reply:
I think adding the test on the size of x is a good solution for the normalization.
The case when the range of x does not contain the peak should be dealt with at the beginning.